src/collector.rs

```diff
@@ -51,6 +51,19 @@ impl Collector {
         self.raw.get(command).unwrap().clone()
     }
 
+    // returns a string representation of the first capture group of the regex
+    pub fn get_regex_from_command(&mut self, command: &str, regex: &str) -> Option<String> {
+        let re = Regex::new(regex).unwrap();
+        let out = self.run_command(command);
+        let output = String::from_utf8(out.stdout).unwrap();
+        let cap = re.captures(&output).unwrap();
```

Comment:
Return either `None` here if keeping return type as `Option<..>`, or return an `Error` here. We should not panic the whole collection process if one collector dies